apps/chat/src/store/application/application.reducers.ts

```diff
@@ -229,6 +231,12 @@ export const applicationSlice = createSlice({
     setHasUnsavedChanges(state, action: PayloadAction<boolean>) {
       state.hasUnsavedChanges = action.payload;
     },
+    setSelectedWidget(state, { payload }: PayloadAction<string>) {
+      state.selectedWidget = payload;
+    },
+    clearSelectedWidget(state) {
+      state.selectedWidget = undefined;
+    },
```

Comment:
I think we can combine these two actions and epics
```suggestion
    setSelectedWidget(state, { payload }: PayloadAction<string | undefined>) {
      state.selectedWidget = payload;
    }
```